src/libraries/System.Private.CoreLib/src/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -145,10 +145,16 @@ public override void Return(T[] array, bool clearArray = false)
                 haveBucket = true;
 
                 // Clear the array if the user requested it.
-                if (clearArray)
+#if DEBUG
```

Comment:
Unconditional clearing here and ignoreing `clearArray` flag is going to be major performance regression.

Reply:
Yes, this can't be merged.

Reply:
it was a typo 🙂